src/Storage/Storage.Management/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -653,151 +654,189 @@ public override void ExecuteCmdlet()
             {
-                createParameters.SasPolicy = new SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"));
+                createContent.SasPolicy = new Track2Models.SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"), null);
+
             }
             if (keyExpirationPeriodInDay != null)
             {
-                createParameters.KeyPolicy = new KeyPolicy(keyExpirationPeriodInDay.Value);
+                createContent.KeyExpirationPeriodInDays = keyExpirationPeriodInDay.Value;
+
             }
-            if(allowCrossTenantReplication != null)
+            if (allowCrossTenantReplication != null)
             {
-                createParameters.AllowCrossTenantReplication = allowCrossTenantReplication;
+                createContent.AllowCrossTenantReplication = this.allowCrossTenantReplication;
             }
             if (this.PublicNetworkAccess != null)
             {
-                createParameters.PublicNetworkAccess = this.PublicNetworkAccess;
+                createContent.PublicNetworkAccess = this.PublicNetworkAccess;
             }
-            if (EnableAccountLevelImmutability.IsPresent || this.immutabilityPeriod != null ||  this.ImmutabilityPolicyState != null)
+            if (EnableAccountLevelImmutability.IsPresent || this.immutabilityPeriod != null || this.ImmutabilityPolicyState != null)
             {
                 if (!EnableAccountLevelImmutability.IsPresent)
                 {
                     throw new ArgumentException("ImmutabilityPeriod, ImmutabilityPolicyState and AllowProtectedAppendWrite, can only be specified with -EnableAccountLevelImmutability.");
                 }
-                createParameters.ImmutableStorageWithVersioning = new ImmutableStorageAccount();
-                createParameters.ImmutableStorageWithVersioning.Enabled = this.EnableAccountLevelImmutability.IsPresent;
+
+                createContent.ImmutableStorageWithVersioning = new Track2Models.ImmutableStorageAccount
+                {
+                    Enabled = this.EnableAccountLevelImmutability.IsPresent
+                };
+
+
                 if (this.immutabilityPeriod != null || this.ImmutabilityPolicyState != null)
                 {
-                    createParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy = new AccountImmutabilityPolicyProperties();
-                    createParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy.ImmutabilityPeriodSinceCreationInDays = this.immutabilityPeriod;
-                    createParameters.ImmutableStorageWithVersioning.ImmutabilityPolicy.State = this.ImmutabilityPolicyState;
+                    createContent.ImmutableStorageWithVersioning.ImmutabilityPolicy = new Track2Models.AccountImmutabilityPolicyProperties
+                    {
+                        ImmutabilityPeriodSinceCreationInDays = this.ImmutabilityPeriod,
+                        State = this.ImmutabilityPolicyState,
+                    };
```

Comment:
New Track2Models.AccountImmutabilityPolicyProperties will fail with following exception, when not set "ImmutabilityPolicyState" in the cmdlet parameter, but set "immutabilityPeriod" .
We should allow user only set "immutabilityPeriod" and not set "ImmutabilityPolicyState" on server. This should be an SDK issue. please record it.

System.ArgumentNullException
HResult=0x80004003